backend/src/main/java/ca/bc/gov/hlth/mohums/controller/UsersController.java

```diff
@@ -51,4 +50,14 @@ public UsersController(WebClientService webClientService) {
         String path = usersPath + "/" + id;
         return webClientService.get(path, null);
     }
+
+    @PostMapping(value = "/users")
+    public Mono<ResponseEntity<Object>> createUser(@RequestBody Object body) {
+        Mono<ClientResponse> post = webClientService.post("https://common-logon-dev.hlth.gov.bc.ca/auth/admin/realms/moh_applications/users", body);
+        return post.flatMap(response -> Mono.just(
+                ResponseEntity.status(response.statusCode())
+                        .headers(response.headers().asHttpHeaders())
+                        .body(response.bodyToMono(Object.class))));
```

Comment:
Originally I used `block` to get the Keycloak response and write some simple imperative code, but it turns out that you can't call `block` in a reactive context. You'll get a runtime exception, something like "block()/blockFirst()/blockLast() are blocking, which is not supported in thread...". 

It seems that the reactive context uses very few threads and guards them carefully, so it prevents code from deliberately blocking. I didn't notice this at first because you're allowed to block when running the test framework (not a reactive context, I guess), but not when you're running the real application.

Fortunately @tschia pointed me in the right direction with some example code from this blog post: https://blog.jdriven.com/2019/10/spring-security-5-2-oauth-2-exploration-part2/

----

There is one business logic problem here. The current code copies all of the header as-is, but all I need (AFAIK) is the Location header, and that header needs to be modified. We don't want to return `Location: https://common-logon-dev.hlth.gov.bc.ca/auth/admin/realms/moh_applications/users/d862b0ee-1e3f-423b-a200-55f2e8f103d9`, we want to return `Location: https://localhost/users/d862b0ee-1e3f-423b-a200-55f2e8f103d9`